packages/lodestar/src/chain/clock/local/LocalClock.ts

```diff
@@ -0,0 +1,80 @@
+import {IBeaconClock} from "../interface";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {computeEpochAtSlot, getCurrentSlot} from "@chainsafe/eth2.0-state-transition";
+import {Epoch, Slot} from "@chainsafe/eth2.0-types";
+
+type NewSlotCallback = (slot: Slot) => void;
+type NewEpochCallback = (epoch: Epoch) => void;
+
+export class LocalClock implements IBeaconClock {
+
+  private readonly config: IBeaconConfig;
+  private readonly genesisTime: number;
+  private currentSlot: number;
+  private isRunning: boolean;
+  private newSlotCallbacks: NewSlotCallback[] = [];
+  private newEpochCallbacks: NewEpochCallback[] = [];
+
+
+  public constructor(config: IBeaconConfig, genesisTime: number) {
+    this.config = config;
+    this.genesisTime = genesisTime;
+    //this assumes clock time is trusted
+    this.currentSlot = getCurrentSlot(this.config, this.genesisTime);
+  }
+
+  public async start(): Promise<void> {
+    this.isRunning = true;
+    const diffTillNextSlot = this.getDiffTillNextSlot();
+    setTimeout(
+      this.updateSlot,
+      diffTillNextSlot
+    );
+  }
+  public async stop(): Promise<void> {
+    this.isRunning = false;
+  }
+
+  public getCurrentSlot(): number {
+    return this.currentSlot;
+  }
+
+  public onNewEpoch(cb: NewEpochCallback): void {
+    if(cb) {
+      this.newEpochCallbacks.push(cb);
```

Comment:
I don't think we should store callbacks here.
It makes it hard to unregister callbacks and it creates an additional self-made event queue (rather than just using the built-in event queue).
We should emit clock events and register event listeners inside the modules that need them